std/getopt.d

```diff
@@ -563,14 +563,20 @@ private void getoptImpl(T...)(ref string[] args, ref configuration cfg,
 
             bool optWasHandled = handleOption(option, receiver, args, cfg, incremental);
 
+            GetOptException requiredNotPassed;
             if (cfg.required && !optWasHandled)
             {
-                throw new GetOptException("Required option " ~ option ~
-                    " was not supplied");
+                requiredNotPassed = new GetOptException("Required option "
+                    ~ option ~ " was not supplied");
             }
             cfg.required = false;
 
-            return getoptImpl(args, cfg, rslt, opts[lowSliceIdx .. $]);
+            getoptImpl(args, cfg, rslt, opts[lowSliceIdx .. $]);
```

Comment:
I don't see getopt being enough of bottleneck anywhere to care about micro-optimizations like tail-call here.